proposals/csharp-9.0/function-pointers.md

```diff
@@ -358,9 +358,14 @@ the attribute:
 
 * It is an error to directly call a method annotated with this attribute from C#. Users must obtain a function pointer to
 the method and then invoke that pointer.
-* It is an error to apply the attribute to anything other than a static method. The C# compiler will mark any non-static
-methods imported from metadata with this attribute as unsupported by the language.
-* It is an error to have managed types as parameters or the return type of a method marked with the attribute.
+* It is an error to apply the attribute to anything other than an ordinary static method or ordinary static local function.
+The C# compiler will mark any non-static methods imported from metadata with this attribute as unsupported by the language.
+* It is an error to have managed types as parameters or the return type of a method marked with the attribute. If the parameter
```

Comment:
I believe that the set of types allowed here should be same as the set of types that it is allowed to take address of in C#.

In other words, the set of types allowed for parameters and return types here should be exactly same as the set of the type for which the following compiles:
```
unsafe static void foo(TYPE x)
{
     void *p = &x;
}
```

Does the description match this?

Reply:
That's the intent. I think I can simplify it a bit by just specifying `unmanaged_type`, but that's the goal of the proposal.